python/openinference-instrumentation/src/openinference/instrumentation/config.py

```diff
@@ -343,10 +353,32 @@ def end(self, end_time: Optional[int] = None) -> None:
         span.end(end_time)
 
 
+class _IdGenerator(IdGenerator):
+    """
+    An IdGenerator that uses a different source of randomness to
+    avoid being affected by seeds set by user application.
+    """
+
+    def generate_span_id(self) -> int:
+        while (span_id := randbits(64)) == INVALID_SPAN_ID:
+            continue
+        return span_id
+
+    def generate_trace_id(self) -> int:
+        while (trace_id := randbits(128)) == INVALID_TRACE_ID:
+            continue
+        return trace_id
+
+
 class OITracer(wrapt.ObjectProxy):  # type: ignore[misc]
     def __init__(self, wrapped: Tracer, config: TraceConfig) -> None:
         super().__init__(wrapped)
         self._self_config = config
+        self._self_id_generator = _IdGenerator()
+
+    @property
+    def id_generator(self) -> IdGenerator:
+        return self._self_id_generator
```

Comment:
A bit confused - how does writing our own ID generator fix this? wouldn't otel be ripe with this problem? How is the implementation above different than the OTEL one that it side-steps the problem?

I'm guessing we confirmed the instrumentation was creating duplicates?

Reply:
it's not affected by `seed` as shown in the unit test

Reply:
> I'm guessing we confirmed the instrumentation was creating duplicates?

not confirmed (because we don't have access), but this eliminates one of the possibilities

Reply:
hmm, I see - it seems sorta unlikely to me that we would have that rampant of a problem due to a seed and it also feels like this inherently would be causing issues for any instrumentation out there.

I'm cool to stamp this PR for some confirmation but it feels like it might not solve the problem.

Reply:
> it feels like it might not solve the problem

Agree. This is just to eliminate one avenue of problems. The rationale derives from the potential severity of the problem, not the confirmed existence of the problem.

Reply:
Got it, sounds good